src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -791,8 +793,9 @@ private static string RemoveSxSPsHomeModulePath(string currentProcessModulePath,
                 {
                     string parentDir = Path.GetDirectoryName(trimedPath);
                     string psExePath = Path.Combine(parentDir, powershellExeName);
+                    string oldExePath = Path.Combine(parentDir, oldPowershellExeName);
                     string psDepsPath = Path.Combine(parentDir, "powershell.deps.json");
```

Comment:
Once you change it to `pwsh` in csproj, I guess the .deps.json file will be changed to `pwsh.deps.json`.
But maybe you want to keep the check for `powershell.deps.json` for now just like the check for old "powershell.exe".

Reply:
Yeah, it did get changed to `pwsh.deps.json`, will add `oldDepsPath`

Reply:
Do we really need the deps.json check?

Reply:
I remember we didn't have the .deps.json check initially and it caused a build break and the fix was to add this deps.json check. Here is the related issue #1744 and #1792

Not sure if that requirement is applicable now.

Reply:
I'll keep it for now